lua/pug/modules/lagdetection.lua

```diff
@@ -95,28 +109,35 @@ u.addHook("Tick", "LagDetection", function()
 
 	if not inTolerance then
 		skips = skips + 1
-		if ( lag.rate > lag.trigger ) or ( skips > lag.skips ) then
-			ServerLog( "LAG: ", lag.rate, "Th: ", lag.threshold )
-			ServerLog( "Average: ", sample.mean, "Skips: ",
-			skips, "/", lag.skips )
+		print("Skips: " .. tostring(skips))
+		print("Rate: " .. tostring(lag.rate))
+		print("Average: " .. tostring(sample.mean))
+
+		-- IF there's casual lag
+		if ( sample.mean <= lag.trigger ) or ( skips >= lag.skips ) then
+			print( "LAG!", "Rate: ", lag.rate, "Trigger: ", lag.trigger )
+			print( "Average: ", sample.mean, "Skips: ", skips, "/", lag.skips ) 
 
 			lag.fClean()
 			lag.timeout = sysTime + lag.cooldown
 
 			PUG:Notify( "pug_lagdetected", 3, 5, nil )
 
-			if ( lag.rate > lag.panic ) or ( skips > lag.pSkips )  then
-				lag.fPanic()
-				PUG:Notify( "pug_lagpanic", 3, 5, nil )
-			end
+        end
+
+        -- If there's a panic
+        if ( sample.mean <= lag.panic ) or ( skips >= lag.pSkips )  then
+            lag.fPanic()
+            PUG:Notify( "pug_lagpanic", 3, 5, nil )
 		end
-	else
-		addSample( lag.rate )
-	end
+    end
+
+    addSample( lag.rate )
```

Comment:
This seemed like an important one.

Previously, `addSample` would only be called if `inTolerance` was true, meaning our averages would only include "good" numbers, which causes a few issues;
1. We can't use the mean tickrate as an accurate comparison because it'll be more of a "baseline" than an "average".
2. If the `inTolerance` is repeatedly false, then we'd get stuck in this loop where we never add new values.

Reply:
Just a quick reply here, I decided not to sample laggy ticks as if the server lags continuously for a long time the cleanup might start thinking that's the norm..

Reply:
However after some thinking, I've decided to try it your way. We'll see how it goes, and if it doesn't work out we can always change it back. 👍